src/Console/Output/Progress/ProgressOutputFactory.php

```diff
@@ -24,20 +27,25 @@ public static function create(string $outputType, OutputContext $context): Progr
             $outputType = ProgressOutputInterface::OUTPUT_TYPE_NONE;
         }
 
-        switch ($outputType) {
-            case ProgressOutputInterface::OUTPUT_TYPE_NONE:
-                return new NullOutput();
+        if (!self::isBuiltInType($outputType)) {
+            throw new \RuntimeException(
+                sprintf(
+                    'Something went wrong, "%s" output type is not supported',
+                    $outputType
+                )
+            );
+        }
 
-            case ProgressOutputInterface::OUTPUT_TYPE_DOTS:
-                return new DotsOutput($context);
+        return ProgressOutputInterface::OUTPUT_TYPE_NONE === $outputType
+            ? new NullOutput()
+            : new DotsOutput($context);
```

Comment:
in general, `switch-case` is using `==` instead of `===` , thus forbidden [same as we forbid `==`. but also, imagine having 20 options inside. ternary or if-elseif-elseif-l.... will be insane.

```php
factoryMap = [
    null: ($context) => new NullOutput(),
    dots: ($context) => new DotsOutput($context),
    ...
];

return factoryMap[$outputType]($context); // O(1)
```

Reply:
That's [what I do](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/6842/files#diff-166b79b3f269a4ecbc711d6f2ae4da497b7448c1bc5ba2d688cd8e94cda0e8e0R27-R32) in #6842 where more output types were added 🙂. Here simple ternary is enough.

Reply:
I would be OK to see that part of refactoring already in refactoring PR, as... purpose of refactoring PR should be refactoring for future needs - otherwise you refactor a bit here, and a bit there.

but non-blocking